routes/auth.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express')
+const router = express.Router()
+const {
+  sgidService,
+  scopes,
+  randomnonce,
+} = require('../lib/sgid-client.service')
+const { formatData } = require('../lib/utils')
+
+router.get('/auth/login/:env?', (req, res) => {
+  const env = req.params.env || 'prod'
+  const authUrl = sgidService[env].authorizationUrl(
+    env,
+    scopes,
+    randomnonce
+  ).url
+  res.redirect(authUrl)
+})
+
+router.get('/callback', async (req, res) => {
+  try {
+    const { code, state } = req.query
+    const { accessToken } = await sgidService[state].callback(code, randomnonce)
+
+    // api response for mobile
+    if (req.useragent.isMobile) {
+      return res.json({ accessToken })
```

Comment:
What is the rationale for making the `userinfo` request in the mobile app instead of on the server? If the mobile app is just acting as a dumb frontend, then all client-related actions should be performed by the backend so that we can adhere to the definition of confidential clients as defined in the [OAuth 2.0 spec section 2.1](https://www.rfc-editor.org/rfc/rfc6749#section-2.1).

This comes back to how we want to define our client profile. In the [same section of the spec](https://www.rfc-editor.org/rfc/rfc6749#section-2.1): 

> A client may be implemented as a distributed set of components, each with a different client type and security context (e.g., a distributed client with both a confidential server-based component and a public browser-based component).  If the authorization server does not provide support for such clients or does not provide guidance with regard to their registration, the client SHOULD register each component as a separate client.
   
In a typical web app flow, the browser-based component isn't considered a client because the flow is:
1. Browser gets user consent
2. Browser performs callback redirect to backend confidential client
3. Backend confidential client retrieves all the necessary tokens before performing a browser redirect to the frontend component

In my imagined mobile app flow, the steps are the roughly the same:
1. Mobile app opens up a mobile browser (in-app, or outside the app?) and starts the OAuth flow to get user consent
2. The mobile browser performs the callback redirect to the backend confidential client
3. Backend confidential client retrieves all the necessary tokens before redirecting back to the mobile app. The redirect relies on a universal link which opens up the mobile app

The main question we need to align on is: 
- What is the client profile that we imagine? Is the mobile frontend considered a client? Or is the flow still primarily a backend confidential client?